### ISSUE TITLE
Promisify all mongoose queries

### ISSUE THREAD

**Issue by catarak** [processing/p5.js-web-editor#521]
#### Nature of issue?

- [X] Existing feature enhancement

#### Feature enhancement details:

Right now, all of the mongoose queries use callbacks instead of the new ES6 promise syntax. This should be updated as nested callbacks are difficult to read.

**Comment by dhruvdutt** [comment]
I'm on it.

**Comment by himanshuc3** [comment]
@dhruvdutt still on it?

**Comment by dhruvdutt** [comment]
Yeah. I've already started with it. Will be pushing by this week.

**Comment by migom6** [comment]
is anyone assigned to this issue?

**Comment by catarak** [comment]
@dhruvdutt started working on it but i don't know how far they got. feel free to start on this!

**Comment by catarak** [comment]
it might be helpful to work on this in pieces, like promisify one controller at a time or something.

**Comment by dhruvdutt** [comment]
Ah! I'm sorry, couldn't get back to this. I would be happy to review a PR.
Best of luck @migom6 :+1:

**Comment by siddhant1** [comment]
Working!

**Comment by catarak** [comment]
this is a big issue, so i recommend making small PR's!

**Comment by shakeabi** [comment]
@catarak As the task is huge, I too would like to contribute to this.

**Comment by meiamsome** [comment]
Are we prefer the promise `.then` chains, or should the code be using `async`/`await`? I would prefer the latter for readability

**Comment by meiamsome** [comment]
Are we prefering the promise `.then` chains, or should the code be using `async`/`await`? I would prefer the latter for readability personally.

**Comment by catarak** [comment]
i definitely prefer async / await! i'll update the issue title.

**Comment by siddhant1** [comment]
We have to change the error handling mechanics with async/await , I have some links , I will post them here

**Comment by catarak** [comment]
There's an in-depth conversation about some of the struggles of using async-await with Express: https://github.com/processing/p5.js-web-editor/pull/973

**TL;DR** It makes sense to include an async middleware like `express-async-handler`, or update to a different Node framework, like Fastify or Koa.

**Comment by yugalkaushik** [comment]
@raclim Can I work on this issue?

**Comment by raclim** [comment]
@yugalkaushik feel free to give it a go! I'd echo an earlier comment on breaking this down into smaller PRs!

**Comment by hxrshxz** [comment]
@raclim  hey I would also like to work on htis can I ? submit a PR as well
